rocks-lib/src/lua/mod.rs

```diff
@@ -0,0 +1,70 @@
+use eyre::Result;
+use std::path::PathBuf;
+use target_lexicon::Triple;
+
+use crate::config::{Config, LuaVersion};
+
+pub struct Lua {
+    pub include_dir: PathBuf,
+    pub lib_dir: PathBuf,
+}
+
+impl Lua {
+    pub fn new(version: &LuaVersion, config: &Config) -> Result<Self> {
+        let output = Self::path(version, config)?;
+
+        if output.exists() {
+            Ok(Lua {
+                include_dir: output.join("include"),
+                lib_dir: output.join("lib"),
+            })
+        } else {
+            let host = Triple::host();
+            let target = &host.to_string();
+            let host_operating_system = &host.operating_system.to_string();
+
+            let (include_dir, lib_dir) = match version {
+                LuaVersion::LuaJIT => {
+                    let build = luajit_src::Build::new()
+                        .target(target)
+                        .host(host_operating_system)
+                        .out_dir(output)
+                        .lua52compat(true)
```

Comment:
question: Should this be configurable?

Reply:
Yeah it should, but that'll require a second `LuaJIT52` version and a separate `--lua-version jit52|luajit52` flag. I'll get going with that then :)